src/main/utility/pod.rs

```diff
@@ -10,29 +12,33 @@
 pub unsafe trait Pod: Copy + 'static {}
 
 /// Convert to a slice of raw bytes.
-pub fn to_u8_slice<T>(slice: &[T]) -> &[u8]
+///
+/// Some bytes may be uninialized if T has padding.
+pub fn to_u8_slice<T>(slice: &[T]) -> &[MaybeUninit<u8>]
 where
     T: Pod,
 {
     // SAFETY: Any value and alignment is safe for u8.
     unsafe {
         std::slice::from_raw_parts(
-            slice.as_ptr() as *const u8,
-            slice.len() * std::mem::size_of::<T>(),
+            slice.as_ptr() as *const MaybeUninit<u8>,
+            slice.len() * std::mem::size_of::<MaybeUninit<T>>(),
         )
     }
 }
 
 /// Convert to a mut slice of raw bytes.
-pub fn to_u8_slice_mut<T>(slice: &mut [T]) -> &mut [u8]
+///
+/// Some bytes may be uninialized if T has padding.
+pub fn to_u8_slice_mut<T>(slice: &mut [T]) -> &mut [MaybeUninit<u8>]
```

Comment:
I think the "mut" version of this function should be "unsafe". For example:

```rust
let mut x = [0u8; 10];
{
    let mut y = to_u8_slice_mut(&mut x);
    y[0] = MaybeUninit::uninit();
}
println!("{}", x[0]);
```

Reply:
Nice catch; done.